tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -7265,13 +7265,33 @@ SpirvEmitter::processIntrinsicInterlockedMethod(const CallExpr *expr,
     spvBuilder.createStore(doExpr(outputArg), toWrite, callExpr->getExprLoc());
   };
 
+  // Set up cxxOpCall if dest is CXXOperatorCallExpr or vector swizzling
+  // of CXXOperatorCallExpr to handle texture buffer correctly.
```

Comment:
You should replace this comment with a much more informative comment. You are matching a very particular pattern. So you should write what that pattern is. Otherwise it's very confusing why you are doing all the casts and why you are getting the castExpr base expression. e.g:

```
// If a vector swizzling of a texture is done as an argument of an interlocked method,
// we need to handle the access to the texture buffer element correctly. For example:
//
//  InterlockedAdd(myRWTexture[index].r, 1);
//
// `-CallExpr
//  |-ImplicitCastExpr
//  | `-DeclRefExpr Function 'InterlockedAdd' 'void (unsigned int &, unsigned int)'
//  |-HLSLVectorElementExpr 'unsigned int' lvalue vectorcomponent r
//  | `-ImplicitCastExpr 'vector<uint, 1>':'vector<unsigned int, 1>' <HLSLVectorSplat>
//  |   `-CXXOperatorCallExpr 'unsigned int' lvalue
```

Reply:
Thanks. Updated.